src/mpid/ch4/shm/ipc/xpmem/xpmem_post.h

```diff
@@ -6,7 +6,13 @@
 #define XPMEM_POST_H_INCLUDED
 
 #include "ch4_impl.h"
+
+#ifdef MPIDI_CH4_SHM_ENABLE_XPMEM
 #include "xpmem_seg.h"
+#define MPIDI_XPMEM_ENABLE_LMT (MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE > -1)
+#else
+#define MPIDI_XPMEM_ENABLE_LMT (0)
+#endif
```

Comment:
We might want to create a new CVAR to control XPMEM on/off rather than using `MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE ` because `MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE ` is the threshold for P2P LMT, not for RMA.

Maybe it is better to rename `MPIDI_XPMEM_ENABLE_LMT ` to `MPIDI_IPC_ENABLE_XPMEM` since for RMA we enable XPMEM also for small windows? Sorry for the bad suggestion on `MPIDI_XPMEM_ENABLE_LMT`...

Reply:
I think the problem checking the threshold in `MPIDI_IPC_mpi_isend` is that that function does not currently have any submodule knowledge. Moving the threshold check the would be even messier (now that I look again at the code). I think we should probably pass to `get_mem_addr` the context (P2P or RMA) so that it can use it to return the right IPC type. What do you think?

Reply:
I really don't have a good solution for this right now. I think passing `MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE` as data size to `MPIDI_IPC_get_mem_attr` for RMA operations is not so bad (I have added a comment in the code).

Reply:
It is not clean to pass the P2P or RMA context to `get_mem_addr` in my opinion. I don't have any good suggestion too. I agree that the current code is not too bad. Let's go with it for now:-) We can always adjust it in future when adding more submodules if we find it is necessary.

Reply:
Oh, I missed this discussion.  We have the window buffer length, right?  Why are we not passing that?